docs/reference/core/runtime.md

```diff
@@ -82,20 +83,243 @@ We can easily access the default `Runtime` to run an effect:
+
+Let's talk about each solution in detail.
+
+## Locally Scoped Runtime Configuration
+
+In ZIO all runtime configurations are inherited from their parent workflows. So whenever we access a runtime configuration, or obtain a runtime inside a workflow, we are accessing the runtime of the parent workflow. We can override the runtime configuration of the parent workflow by providing a new configuration to a region of the code. This is called locally scoped runtime configuration. When the execution of that region is finished, the runtime configuration will be restored to its original value.
+
+We mainly use `ZIO#provideXYZ` operators to provide a new runtime configuration to a specific region of the code:
+
+### Configuring Runtime by Providing Configuration Layers
+
+By providing (`ZIO#provideXYZ`) runtime configuration layers to a ZIO workflow, we can change the runtime configs easily:
+
+```scala mdoc:compile-only
+import zio._
+
+object MainApp extends ZIOAppDefault {
+  val addSimpleLogger: ZLayer[Any, Nothing, Unit] =
+    Runtime.addLogger((_, _, _, message: () => Any, _, _, _, _) => println(message()))
+
+  def run = {
+    for {
+      _ <- ZIO.log("Application started!")
+      _ <- ZIO.log("Application is about to exit!")
+    } yield ()
+  }.provide(Runtime.removeDefaultLoggers ++ addSimpleLogger)
+}
+```
+
+The output:
+
+```scala
+Application started!
+Application is about to exit!
+```
+
+To provide runtime configuration to a specific region of a ZIO application, we should provide the configuration layer only to that specific region:
+
+```scala mdoc:compile-only
+import zio._
+
+object MainApp extends ZIOAppDefault {
+  val addSimpleLogger: ZLayer[Any, Nothing, Unit] =
+    Runtime.addLogger((_, _, _, message: () => Any, _, _, _, _) => println(message()))
+
+  def run =
+    for {
+      _ <- ZIO.log("Application started!")
+      _ <- {
+        for {
+          _ <- ZIO.log("I'm not going to be logged!")
+          _ <- ZIO.log("I will be logged by the simple logger.").provide(addSimpleLogger)
+          _ <- ZIO.log("Reset back to the previous configuration, so I won't be logged.")
+        } yield ()
+      }.provide(Runtime.removeDefaultLoggers)
+      _ <- ZIO.log("Application is about to exit!")
+    } yield ()
+}
+```
+
+The output:
+
+```scala
+timestamp=2022-08-31T14:28:34.711461Z level=INFO thread=#zio-fiber-6 message="Application started!" location=<empty>.MainApp.run file=ZIOApp.scala line=9
+I will be logged by the simple logger.
+timestamp=2022-08-31T14:28:34.832035Z level=INFO thread=#zio-fiber-6 message="Application is about to exit!" location=<empty>.MainApp.run file=ZIOApp.scala line=17
+```
+
+### Configuring Runtime Using `bootstrap` Layer
+
+The `bootstrap` layer is a special layer that is mainly used to acquiring and releasing services that are necessary for the application to run. However, it can also be applied to runtime customization as well.
```

Comment:
It would be good to clarify that the `bootstrap` layer is only relevant when extending `ZIOApp`.